src/Kafka/Consumer.purs

```diff
@@ -210,9 +233,10 @@ type EachBatchHandlerImpl =
 
 -- | see [Each Batch](https://kafka.js.org/docs/consuming#a-name-each-batch-a-eachbatch)
 -- |
+-- | * `commitOffsets`
+-- |   * commits offsets if provided, otherwise commits most recently resolved offsets ignoring [autoCommit](https://kafka.js.org/docs/consuming#auto-commit) configurations (`autoCommitInterval` and `autoCommitThreshold`)
```

Comment:
❓ If the behavior of this function changes depending on the presence of offsets, shouldn't we have a separate function for committing `uncommittedOffsets`?

Reply:
Before using it in production and see how often each function is used I try not to deviate too far from the `kafkajs` library. `kafkajs` doesn't expose `commitOffsets` but there is a `commitOffsets` on `consumerGroup`.

https://github.com/tulios/kafkajs/blob/196105c224353113ae3e7f8ef54ac9bad9951143/src/consumer/runner.js#L308-L312